spec/rspec/core/bisect/runner_spec.rb

```diff
@@ -213,6 +213,27 @@ def repro_command_from(ids)
       open3_method = Open3.respond_to?(:capture2e) ? :capture2e : :popen3
       open3_method = :popen3 if RSpec::Support::Ruby.jruby?
 
+      RSpec::Matchers.define :invoke_command_with_env do |command, environment|
+        match do |actual|
+          if open3_method == :capture2e
+            actual.call
+            expect(Open3).to have_received(open3_method).with(environment, command)
+          elsif open3_method == :popen3
+            called_environment = nil
+            allow(Open3).to receive(open3_method) do
+              called_environment = ENV.to_hash.dup
+            end
```

Comment:
Ah, good point - I've cleaned up the stubbing. I'm probably being daft here but I can't figure out what the assertion would look like in non-block form, though (I'm unsure what the "actual" value would be to read well):

```ruby
it "runs the command" do
  runner.original_results

  expect(???).to have_invoked_command_with_env(a_string_including("--seed 1234"), {})
end
```

`expect(a_string_including("--seed 1234").to have_been_invoked_with_env({})` perhaps?